src/DefaultUser.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-authentication for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-authentication/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Authentication;
+
+class DefaultUser implements UserInterface
```

Comment:
I would make this class `final`, and mark the properties as `private`. The interface is pretty slim, and it's less likely that a developer will extend this class than provide a clean implementation.